ambry-messageformat/src/main/java/com/github/ambry/messageformat/MessageFormatSend.java

```diff
@@ -129,7 +130,7 @@ private void fetchDataFromReadSet(MessageFormatMetrics metrics) throws MessageFo
           // If we can successfully deserialize BlobAll, then the message content is correct
           try {
             ByteBuf messageData = readSet.getPrefetchedData(i).duplicate();
-            MessageFormatRecord.deserializeBlobAll(new ByteBufInputStream(messageData), storeKeyFactory);
+            MessageFormatRecord.deserializeBlobAll(new NettyByteBufDataInputStream(messageData), storeKeyFactory);
```

Comment:
What's the reason that we want to change the InputStream type?

Reply:
to avoid one extra copy. If we use NettyByteBufDataInputStream, we don't have to copy 4MB of data chunk out to a java ByteBuffer to to calculate crc.